compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -1874,6 +1875,79 @@ struct DescendingSortX86DataSnippetByDataSize
       return a->getDataSize() > b->getDataSize();
       }
    };
+
+
+void OMR::X86::CodeGenerator::addItemsToRSSReport(uint8_t *oldColdCode, uint8_t *coldCode)
+   {
+   // calculate size and collect counters for all cold blocks in the cold cache
+   TR_PersistentList<TR::DebugCounterAggregation> coldBlockCountersList;
+   TR_PersistentList<TR::DebugCounterAggregation> emptyCountersList;
+   bool insideColdCode = false;
+   size_t blocksInsideColdCodeSize = 0;
+   const char* methodName = self()->comp()->signature();
+
+   for (TR::TreeTop *tt = self()->comp()->getMethodSymbol()->getFirstTreeTop(); tt ; tt = tt->getNextTreeTop())
+      {
+      TR::Node *node = tt->getNode();
+
+      if (node->getOpCodeValue() != TR::BBStart) continue;
+
+      TR::Block *block;
+
+      block = node->getBlock();
+
+      if (insideColdCode)
+         {
+         uint8_t *startCursor = block->getFirstInstruction()->getBinaryEncoding();
+         uint8_t *endCursor = block->getLastInstruction()->getBinaryEncoding();
+         size_t blockSize = endCursor - startCursor;
+         blocksInsideColdCodeSize += blockSize;
+
+         coldBlockCountersList.add(block->getDebugCounters());
+         }
+
+      if (block->isLastWarmBlock())
+         insideColdCode = true;
+
+      }
+
+   // create one RSSItem for all cold blocks
```

Comment:
Why create an RSSItem for cold blocks if there's going to be an RSS item for the cold code in `OMRCodeCache`?

Reply:
`OMRCodeCache` contains an `RSSRegion` which represents the whole cold part of that cache. `RSSItem` describes some part of the region. There can be many `RSSItem`s in one region. For example, cold code cache can contain cold blocks, snippets, OOL code. I tried to describe that in `OMRRSSReport.hpp`.